src/dc_receive_imf.rs

```diff
@@ -155,6 +137,23 @@ pub fn dc_receive_imf(
         }
     }
 
+    for header_def in vec![HeaderDef::To, HeaderDef::Cc] {
+        if let Some(field) = mime_parser.get(header_def) {
+            dc_add_or_lookup_contacts_by_address_list(
+                context,
+                &field,
+                if !incoming {
+                    Origin::OutgoingTo
+                } else if incoming_origin.is_verified() {
+                    Origin::IncomingTo
+                } else {
+                    Origin::IncomingUnknownTo
+                },
+                &mut to_ids,
+            )?;
+        }
+    }
+
```

Comment:
To and CC have no big semantic difference and there was no good reason to split To and CC parsing to two functions. 
If anyone has a suggestion on how to avoid `vec![...]` please let me know.  i couldn't convince the rust compiler without it.

Reply:
you should be able to write `&[a, b]` instead which uses an array